appium/lib/screen-object/accessors/element.rb

```diff
@@ -81,26 +96,116 @@ def dynamic_text_exists? dynamic_text
         end
       end
 
-      def scroll
-        $driver.execute_script 'mobile: scrollTo',:element => element.ref
-        # $driver.execute_script("mobile: scroll",:direction => direction.downcase, :element => element.ref)
+      def gesture(args)
+        Appium::TouchAction.new($driver).swipe(start_x: args[0], start_y: args[1], end_x: args[2], end_y: args[3], duration: args[4]).perform
+      rescue RuntimeError => e
+        raise("Error during gesture in element.rb\n Error Details: #{e}")
       end
 
-      def scroll_to_text(text)
-        $driver.scroll_to(text)
+      def scroll(direction = :down, duration = 1000)
+        size = driver.window_size
+        x = size.width / 2
+        y = size.height / 2
+        loc = (case direction
+               when :up then    [x, y * 0.5, x, (y + (y * 0.3)), duration]
+               when :down then  [x, y, x, y * 0.5, duration]
+               when :left then  [x * 0.6, y, x * 0.3, y, duration]
+               when :right then [x * 0.3, y, x * 0.6, y, duration]
+               else
+                 raise('Only upwards, downwards, leftwards and rightwards scrolling are supported')
+               end)
+        gesture(loc)
       end
 
-      def scroll_to_exact_text(text)
-        $driver.scroll_to_exact(text)
+      def wait_until(timeout = 5, message = nil, &block)
+        driver.no_wait
+        wait = Selenium::WebDriver::Wait.new(timeout: timeout, message: message)
+        wait.until &block
+        driver.default_wait
       end
 
-      def scroll_for_element_click
+      def element_visible?(direction = :down)
+        p default_wait = driver.default_wait
+        driver.no_wait
         if element.displayed?
-          element.click
+          driver.set_wait(default_wait)
+          true
         else
-          scroll
-          element.click
+          scroll(direction)
+          false
         end
+      rescue
+        scroll(direction)
+        false
+      end
+
+      def scroll_element_to_view(direction = :down, time_out = 30)
+        wait_until(time_out,'Unable to find element',&->{element_visible?(direction)})
+        scroll(direction)
+      end
+
+      def scroll_element_to_view_click(direction= :down, time_out = 40)
+        wait_until(time_out,'Unable to find element',&->{element_visible?(direction)})
+        scroll(direction)
+        element.click
+      end
+
+      def scroll_down
+        scroll(:down)
```

Comment:
Update all these method calls from `scroll(:down)` to `ScreenObject::Accessors.scroll(:down)`